xmodule/util/keys.py

```diff
@@ -28,6 +28,14 @@ class BlockKey(NamedTuple):
     def from_usage_key(cls, usage_key):
         return cls(usage_key.block_type, usage_key.block_id)
 
+    def __str__(self):
+        return f"{self.type}@{self.id}"
+
+    @classmethod
+    def from_string(cls, s):
```

Comment:
let's annotate this:

```suggestion
    def from_string(cls, s: str) -> Self:
```

where `Self` is part of `typing`.

If that doesn't work, could also annotate as `-> "BlockKey":`